go/kbfs/libkbfs/favorites.go

```diff
@@ -336,6 +357,11 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
 			}
 			// If we weren't explicitly asked to refresh, we can return possibly
 			// stale favorites rather than return nothing.
+			if err == context.DeadlineExceeded {
+				newCtx, _ := context.WithTimeout(context.Background(),
+					favoritesBackgroundRefreshTimeout)
+				go f.RefreshCache(newCtx)
```

Comment:
I'm worried that this could trigger a bunch of goroutines.

Reply:
RefreshCache checks whether there are any other refreshes currently running before it does anything. 

	select {
	case f.refreshWaiting <- struct{}{}:
	default:
		// There is already a refresh in the queue
		return
	}